pipeline/frontend/metadata/environment.go

```diff
@@ -0,0 +1,169 @@
+// Copyright 2023 Woodpecker Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metadata
+
+import (
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+var pullRegexp = regexp.MustCompile(`\d+`)
+
+// Environ returns the metadata as a map of environment variables.
+func (m *Metadata) Environ() map[string]string {
+	var (
+		repoOwner    string
+		repoName     string
+		sourceBranch string
+		targetBranch string
+	)
+
+	repoParts := strings.Split(m.Repo.Name, "/")
+	if len(repoParts) == 2 {
+		repoOwner = repoParts[0]
+		repoName = repoParts[1]
+	} else {
+		repoName = m.Repo.Name
```

Comment:
Note: I think it should be [repoOwner, ...repoName]

Reply:
I meant vice versa: repo-name just the last part and owner the rest

Reply:
I think we should extend that in the metadata itselve: -> name, owner, id ...

Reply:
*ForgeID

Reply:
or RemoteID ?